modules/lang-painless/src/test/java/org/elasticsearch/painless/AnalysisScriptTests.java

```diff
@@ -0,0 +1,32 @@
+package org.elasticsearch.painless;
+
+import org.elasticsearch.painless.spi.Whitelist;
+import org.elasticsearch.script.AnalysisPredicateScript;
+import org.elasticsearch.script.ScriptContext;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AnalysisScriptTests extends ScriptTestCase {
+
+    @Override
+    protected Map<ScriptContext<?>, List<Whitelist>> scriptContexts() {
+        Map<ScriptContext<?>, List<Whitelist>> contexts = new HashMap<>();
+        contexts.put(AnalysisPredicateScript.CONTEXT, Whitelist.BASE_WHITELISTS);
+        return contexts;
+    }
+
+    public void testAnalysisScript() {
+        AnalysisPredicateScript.Factory factory = scriptEngine.compile("test", "return \"one\".contentEquals(term.term)",
+            AnalysisPredicateScript.CONTEXT, Collections.emptyMap());
```

Comment:
Having to use `contentEquals` is a bit trappy here as well, but I want to avoid creating Strings on every call to `incrementToken()`.